plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+        flb_sds_len_set(tag, 0);
+        flb_sds_cat(tag, ctx->ins->tag, flb_sds_len(ctx->ins->tag));
+    }
+    else {
+        /* Compose the query string using the URI */
+        len = strlen(uri);
+
+        if (len == 1) {
+            tag = NULL; /* use default tag */
+        }
+        else {
+            tag = flb_sds_create_size(len);
+            if (!tag) {
+                mk_mem_free(uri);
+                return -1;
+            }
+
+            /* New tag skipping the URI '/' */
+            flb_sds_cat(tag, uri + 1, len - 1);
+
+            /* Sanitize, only allow alphanum chars */
+            for (i = 0; i < flb_sds_len(tag); i++) {
+                if (!isalnum(tag[i]) && tag[i] != '_' && tag[i] != '.') {
+                    tag[i] = '_';
+                }
+            }
+        }
+    }
+
+    /* Check if we have a Host header: Hostname ; port */
+    mk_http_point_header(&request->host, &session->parser, MK_HEADER_HOST);
+
+    /* Header: Connection */
+    mk_http_point_header(&request->connection, &session->parser,
+                         MK_HEADER_CONNECTION);
+
+    /* HTTP/1.1 needs Host header */
+    if (!request->host.data && request->protocol == MK_HTTP_PROTOCOL_11) {
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return -1;
+    }
+
+    /* Should we close the session after this request ? */
+    mk_http_keepalive_check(session, request, ctx->server);
+
+    /* Content Length */
+    header = &session->parser.headers[MK_HEADER_CONTENT_LENGTH];
+    if (header->type == MK_HEADER_CONTENT_LENGTH) {
+        request->_content_length.data = header->val.data;
+        request->_content_length.len  = header->val.len;
+    }
+    else {
+        request->_content_length.data = NULL;
+    }
+
+    if (request->method == MK_METHOD_GET) {
+        /* Handle health minotoring of splunk hec endpoint for load balancers */
+        if (strncmp(uri, "/services/collector/health", 26) == 0) {
+            send_json_message_response(conn, 200, "{\"text\":\"Success\",\"code\":200}");
+        }
+        else {
+            send_response(conn, 400, "error: invalid HTTP endpoint\n");
+        }
+
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return 0;
+    }
+
+    /* Under services/collector endpoints are required for
+     * authentication if provided splunk_token */
+    ret = validate_auth_header(ctx, request);
+    if (ret < 0){
+        send_response(conn, 401, "error: unauthroized\n");
+        if (ret == SPLUNK_AUTH_MISSING_CRED) {
+            flb_plg_warn(ctx->ins, "missing credentials in request headers", ret);
+        }
+        else if (ret == SPLUNK_AUTH_UNAUTHORIZED) {
+            flb_plg_warn(ctx->ins, "wrong credentials in request headers", ret);
+        }
+
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return -1;
+    }
+
+    if (request->method == MK_METHOD_POST) {
+        if (strncmp(uri, "/services/collector/event", 25) == 0 ||
```

Comment:
Same as 745

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/7349/commits/e1eb2a517eae3d17099d8e1336770225bb42d290.